.grit/patterns/js/intelligent_useEffect_to_useLayoutEffect.md

```diff
@@ -0,0 +1,109 @@
+---
+title: [React2Hooks] Intelligent useEffect vs useLayoutEffect
+---
+
+If a useEffect depends on layout etc. it should switch to useLayoutEffect.
+
+tags: #fix
+
+```grit
+engine marzano(0.1)
+language js
+
+
+or {
+
+ `useEffect(() => {
+    $body 
+    },[$conditions])` as $effectHook where {
```

Comment:
To make this more general for different function call styles I'd usually use:

```suggestion
 `useEffect($body,[$conditions])` as $effectHook where {
```

Reply:
Hi @morgante, really helped,  the pattern become more generic